smokefree/sites/all/modules/custom/icfi_api_keys/ICFIApi.inc

```diff
@@ -0,0 +1,104 @@
+<?php
+
+
+class ICFIApi {
+  protected $machineName;
+  protected $apiValue;
+  protected $isJsVariable;
+
+  public function __construct($machineName, $apiValue, $isJsVariable) {
+    if (empty($machineName) || empty($apiValue)) {
+      throw new InvalidArgumentException('Please supply a valid api value and an appropriate machine name');
+    }
+
+    // Remove accidental whitespaces at the end
+    $this->machineName = trim($machineName);
+    $this->apiValue = trim($apiValue);
+    $this->isJsVariable = $isJsVariable;
+  }
+
+  /**
+   * Create a new API key entry in the database
+   */
```

Comment:
Since these are not hooks, @return value should be included in comments
https://www.drupal.org/docs/develop/coding-standards/api-documentation-and-comment-standards#drupal